pkg/gateway/model_build_targets.go

```diff
@@ -111,19 +112,38 @@ func (t *latticeTargetsModelBuildTask) buildLatticeTargets(ctx context.Context)
 		return errors.New(errmsg)
 	}
 
```

Comment:
Nit: We have a giant block `if svc.DeletionTimestamp.IsZero() {}`, I think we can do a reverse condition and return earlier? e.g.,:

```
	if !svc.DeletionTimestamp.IsZero() {
		//build model model for targets deregistering
		t.latticeTargets = latticemodel.NewTargets(t.stack, tgName, latticemodel.TargetsSpec{
			Name:         t.tgName,
			Namespace:    t.tgNamespace,
			RouteName:    t.routename,
			TargetIPList: []latticemodel.Target{},
		})
	}
```

Reply:
I think the current structure has a similar effect no? Since we'll still need to build the resulting model in both cases, regardless if the timestamp is zero or now.

Reply:
current structure has have same logic, I just think our if else loop condition too deep